extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -0,0 +1,229 @@
+						->escaped(),
+				'userErrorMessage' => wfMessage( 'discussionslog-no-user-match-error' )
+						->escaped(),
+				'hasNoUserLogRecords' => $hasNoUserLogRecords,
+				'noUserLogRecordsMessage' => wfMessage( 'discussionslog-no-mobile-activity-error' )
+						->params( $userName )
+						->escaped(),
+				'timestampHeader' => wfMessage( 'discussionslog-timestamp-header' )
+						->escaped(),
+				'userAgentHeader' => wfMessage( 'discussionslog-user-agent-header' )
+						->escaped(),
+				'userLogRecords' => $displayedUserLogRecords
+			]
+		);
+	}
+
+	private function aggregateLogSearches( $userId, $userName ) {
+		$logger = Wikia\Logger\WikiaLogger::instance();
+		$query = $this->getQuery( $userId );
+		$client = new GuzzleHttp\Client();
+		$records = [];
+		$dayOffset = 0;
+		while ( count( $records ) < self::PAGINATION_SIZE && $dayOffset < self::DAYS_RANGE ) {
+			$url = $this->constructKibanaUrl( $dayOffset++ );
+			try {
+				$response = $client->post(
+					$url, [
+						'body' => $query,
+					]
+				);
+			} catch ( \GuzzleHttp\Exception\RequestException $requestException ) {
+				$logger->warning(
+					sprintf( 'Request to elasticsearch failed: %s', $requestException->getMessage() )
+				);
+				break;
+			}
+
+			if ( $response->getStatusCode() !== self::HTTP_STATUS_OK ) {
+				$logger->warning(
+					sprintf( 'Elasticsearch request error; status code %d', $response->getStatusCode() )
+				);
+				break;
+			}
+
+			$resultObject = json_decode( $response->getBody() );
+			$hits = $resultObject->hits->hits;
+
+			foreach ( $hits as $hit ) {
+				$record = $hit->_source;
+
+				$userLogRecord = new UserLogRecord();
+				$userLogRecord->app =
+					$record->{'mobile_app.data.app_name'} . ' ' . $record->{'mobile_app.data.app_version'};
+				$userLogRecord->ip = $record->{'mobile_app.client_ip'};
+				$userLogRecord->language = $record->{'mobile_app.event.device_language'};
+				$userLogRecord->location =
+					$record->{'mobile_app.geo_ip.city'} . ', ' . $record->{'mobile_app.geo_ip.country_name'};
+				$userLogRecord->timestamp = date( DATE_RFC2822, $record->{'mobile_app.event.timestamp'} / 1000 );
+				$userLogRecord->userAgent =
+					$record->{'mobile_app.data.platform'} . ' ' . $record->{'mobile_app.data.platform_version'};
+				$userLogRecord->userId = $userId;
+				$userLogRecord->userName = $userName;
+				$records[] = $userLogRecord;
+			}
+		}
+
+		return $records;
+	}
+
+	private function getQuery( $userId ) {
+		$pagination_size = self::PAGINATION_SIZE;
+		return <<<JSON_BODY
+{
+	"query": {
+		"filtered": {
+			"query": {
+				"bool": {
+					"should": [{
+						"query_string": {
+							"query":"mobile_app.event.user_id:$userId"
+						}
+					}]
+				}
+			}
+		}
+	},
+	"size":$pagination_size,
+	"sort":[{
+		"@timestamp": {
+			"order": "desc"
+		}
+	}]
+}
+JSON_BODY;
+	}
+
+	private function checkAccess() {
+		if ( !$this->wg->User->isLoggedIn() || !$this->wg->User->isAllowed( 'forumadmin' ) ) {
```

Comment:
Right, it should use 'specialdiscussionslog' permissions, good catch! And sorry for that ;)